### DISCUSSION TITLE
Creating an fp-ts GitHub Organization

### DISCUSSION THREAD

**Discussion by theurgi** [gcanti/fp-ts#1802] [General]
@gcanti 

Having a quick look around, it seems that `fp-ts` has grown to be the frontrunner in terms of popularity and development among functional TypeScript libraries.

Has creating a GitHub/npm organization to consolidate associated codebases and increase visibility and cohesion of the broader project been considered? Another thing to consider is setting up an [open collective](https://opencollective.com/).

I know this would bring additional overhead in the short term, but doing so may invite more contribution (both technical and financial) to support the longevity of this project (and its core maintainer 🙂) and streamline maintenance as it continues to grow:

![graph of fp-ts downloads from npmtrends.com](https://user-images.githubusercontent.com/58859663/198869156-c3fd2b30-eddf-4c47-9eff-feb759c7640a.png)

**Comment by gcanti** [comment]
Hi @theurgi, 

about the org take a look at https://github.com/fp-ts (here's the [plan](https://github.com/fp-ts/core#typed-functional-programming-in-typescript))
about sponsorship my GitHub Sponsors profile is live (both on my personal account and the org)

**Comment by theurgi** [comment]
Ahh! Two steps ahead of me 😅. Well, awesome! Excited for the future of this project.

After giving the new plan a quick look—and maybe this is an issue to open there—but have you considered a monorepo? Is there a particular reason you've opted not to use one?

Something like this: 

```
fp-ts
├──📂 .changeset
├──📂 .vscode
├──📂 docs
│  ├──📂 api-reference
│  ├──📂 intro-to-fp-book
│  └──📂 web
├──📂 packages
│  ├──📂 codec
│  ├──📂 config
│  │  ├──📂 eslint-config
│  │  └──📂 tsconfig
│  │     ├── base.json
│  │     ├── build.json
│  │     ├── build-test.json
│  │     └── eslint.json
│  ├──📂 core
│  ├──📂 data
│  ├──📂 optics
│  └──📂 etc...
├── package.json
├── pnpm-lock.yaml
├── pnpm-workspace.yaml
├── .eslintrc.js
├── tsconfig.json
└── etc ...
```

**Reply by gcanti** [reply]
In this early phase /core and /data are separated just because is more handy but we will consider a monorepo as soon as /core will be more stable
